### PR TITLE
Bmd 5126

### PR DESCRIPTION
**Pull request by bmwosz**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -252,7 +252,81 @@ resource "aws_cloudwatch_log_group" "app" {
...
+  validation_method         = "DNS"
+}
 
+resource "aws_route53_record" "alb_cert_validation" {
+  count   = (var.alb_internal_create || var.alb_external_create) && var.alb_main_domain != "" ? 1 : 0
+  name    = aws_acm_certificate.alb_cert.0.domain_validation_options.0.resource_record_name
+  type    = aws_acm_certificate.alb_cert.0.domain_validation_options.0.resource_record_type
+  zone_id = data.aws_route53_zone.zone.zone_id
+  records = [aws_acm_certificate.alb_cert.0.domain_validation_options.0.resource_record_value]
+  ttl     = 60
```
[the remark this answers is not in the document]
Comment (bmwosz) [reply main.tf:275]: In all cases I used this resource that was enough.
But we can change that value to ex. 5 minutes (300)

File: versions.tf [hunk 2]
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = "~> 0.12.24"
+  required_version = "> 0.12.24"
```
[the remark this answers is not in the document]
Comment (bmwosz) [reply versions.tf:2]: I have added the same version as we started working in terraform.
on 0.13 update we will need to change that value also.

File: main.tf [hunk 3]
```diff
@@ -252,7 +252,81 @@ resource "aws_cloudwatch_log_group" "app" {
...
+  internal                                = true
+  http_enabled                            = true
+  http_redirect                           = true
+  https_enabled                           = true
+  certificate_arn                         = aws_acm_certificate.alb_cert[0].arn
+  access_logs_enabled                     = false
+  alb_access_logs_s3_bucket_force_destroy = true
+  access_logs_region                      = var.region
+  cross_zone_load_balancing_enabled       = true
+  http2_enabled                           = true
```
[the remark this answers is not in the document]
Comment (bmwosz) [reply main.tf:304]: I don't believe that this change will add anything.

File: main.tf [hunk 4]
```diff
@@ -252,7 +252,81 @@ resource "aws_cloudwatch_log_group" "app" {
...
+  internal                                = false
+  http_enabled                            = true
+  http_redirect                           = true
+  https_enabled                           = true
+  certificate_arn                         = aws_acm_certificate.alb_cert[0].arn
+  access_logs_enabled                     = false
+  alb_access_logs_s3_bucket_force_destroy = true
+  access_logs_region                      = var.region
+  cross_zone_load_balancing_enabled       = true
+  http2_enabled                           = true
```
[the remark this answers is not in the document]
Comment (bmwosz) [reply main.tf:328]: I don't believe that this change will add anything.

File: main.tf [hunk 5]
```diff
@@ -252,7 +252,81 @@ resource "aws_cloudwatch_log_group" "app" {
...
+  certificate_arn         = aws_acm_certificate.alb_cert.0.arn
+  validation_record_fqdns = [aws_route53_record.alb_cert_validation.0.fqdn]
+}
+
+# ALB
+
+module "alb_default_internal" {
+  source                                  = "git::https://github.com/cloudposse/terraform-aws-alb.git?ref=tags/0.18.0"
+  namespace                               = var.namespace
+  name                                    = "${var.name}-i"
```
[the remark this answers is not in the document]
Comment (bmwosz) [reply main.tf:289]: Good idea to create additional variables, but I would like not to set them in default.
I would rather prefere to use first letter of `name` a `namespace` as short names if variables are empty.
What do you think @Maciej-Gierwatowski ?

[the remark this answers is not in the document]
Comment (Maciej-Gierwatowski) [reply main.tf:289]: Using first letters may quickly throw us into unexpected conflicts with other namespaces/names

File: main.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (Maciej-Gierwatowski) [reply main.tf:304]: As we found out, currently we can't disable ALB module, therefore we are forced to create ALBs. However we can at least prevent creating listeners on those ALBs

File: main.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (Maciej-Gierwatowski) [reply main.tf:275]: I don't have opinion on how long should be TTL, just pointed this **might** be too short.

**Review by bmtoml — APPROVED** [review]
lgtm

**Pull request merged**